File: WinForms/VB/DialogsMainDemo/CodeExamples/SvgPreviewHandler.vb
' Assembly {TutorialsAssembly}
Imports DevExpress.Dialogs.Core.Items
Imports DevExpress.Utils.Svg
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Base.Controls.Preview
Imports DevExpress.XtraEditors.Controls
Imports System.Windows.Forms

Namespace DevExpress.DXperience.Demos.CodeDemo.Data

    Public Class SvgPreviewHandler
        Inherits DevExpress.XtraEditors.Base.Controls.Preview.WinPreviewHandlerBase

        Private svgImage As DevExpress.Utils.Svg.SvgImage

        Public Overrides Function Load(ByVal file As String, ByVal shellItem As DevExpress.Dialogs.Core.Items.ShellItem) As Boolean
            MyBase.Load(file, shellItem)
            Dim loaded As Boolean = False
            If Equals(shellItem.Extension.ToLower(), ".svg") Then
                Try
                    Me.svgImage = DevExpress.Utils.Svg.SvgImage.FromFile(file)
                    loaded = True
                Catch
                End Try
            End If

            Return loaded
        End Function

        Public Overrides Function CreatePreviewControl() As Control
            Dim picEdit = New DevExpress.XtraEditors.PictureEdit()
            picEdit.SvgImage = Me.svgImage
            picEdit.Enabled = False
            picEdit.BorderStyle = DevExpress.XtraEditors.Controls.BorderStyles.NoBorder
            picEdit.Properties.SizeMode = DevExpress.XtraEditors.Controls.PictureSizeMode.Squeeze
            picEdit.Properties.UseDisabledStatePainter = False
            Return picEdit
        End Function
    End Class
End Namespace

File: WinForms/CS/DialogsMainDemo/CodeExamples/SvgPreviewHandler.cs
// Assembly {TutorialsAssembly}
namespace DevExpress.DXperience.Demos.CodeDemo.Data {
    using DevExpress.Dialogs.Core.Items;
    using DevExpress.Utils.Svg;
    using DevExpress.XtraEditors;
    using DevExpress.XtraEditors.Base.Controls.Preview;
    using DevExpress.XtraEditors.Controls;
    using System.Windows.Forms;

    public class SvgPreviewHandler : WinPreviewHandlerBase {
        SvgImage svgImage;

        public override bool Load(string file, ShellItem shellItem) {
            base.Load(file, shellItem);
            bool loaded = false;
            if(shellItem.Extension.ToLower() == ".svg") {
                try {
                    this.svgImage = SvgImage.FromFile(file);
                    loaded = true;
                }
                catch { }
            }
            return loaded;
        }
        public override Control CreatePreviewControl() {
            var picEdit = new PictureEdit();
            picEdit.SvgImage = svgImage;
            picEdit.Enabled = false;
            picEdit.BorderStyle = BorderStyles.NoBorder;
            picEdit.Properties.SizeMode = PictureSizeMode.Squeeze;
            picEdit.Properties.UseDisabledStatePainter = false;
            return picEdit;
        }
    }
}
